Description: [orcus] use the foreground color for solid fill
  See https://gitlab.com/orcus/orcus/commit/f821995022df8dd1e580dd22cf131584b2b1ac4f
  and https://gitlab.com/orcus/orcus/commit/f7953a814d6a43205791b6cc01c528ef5d4b1ce3
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug: https://bugs.documentfoundation.org/show_bug.cgi?id=115931
Bug-Ubuntu: https://launchpad.net/bugs/1750982
Forwarded: no

--- a/sc/source/filter/orcus/interface.cxx
+++ b/sc/source/filter/orcus/interface.cxx
@@ -839,7 +839,7 @@ void ScOrcusStyles::fill::applyToItemSet
         return;
     }
 
-    rSet.Put(SvxBrushItem(maBgColor, ATTR_BACKGROUND));
+    rSet.Put(SvxBrushItem(maFgColor, ATTR_BACKGROUND));
 }
 
 ScOrcusStyles::protection::protection():
--- a/sc/qa/unit/subsequent_filters-test.cxx
+++ b/sc/qa/unit/subsequent_filters-test.cxx
@@ -2395,7 +2395,7 @@ void ScFiltersTest::testOrcusODSStyleInt
     CPPUNIT_ASSERT_MESSAGE("Style Name1 : Doesn't have Attribute background, but it should have.",
         pStyleSheet->GetItemSet().HasItem(ATTR_BACKGROUND, &pItem));
     const SvxBrushItem* pBackground = static_cast<const SvxBrushItem*>(pItem);
-    CPPUNIT_ASSERT_EQUAL(Color(254, 255, 204), pBackground->GetColor());
+    CPPUNIT_ASSERT_EQUAL(Color(255, 254, 255, 204), pBackground->GetColor());
 
     CPPUNIT_ASSERT_MESSAGE("Style Name1 : Doesn't have Attribute border, but it should have.",
         pStyleSheet->GetItemSet().HasItem(ATTR_BORDER, &pItem));
